ui/src/views/DNSRecords.vue

```diff
@@ -0,0 +1,388 @@
+      },
+      error: {
+        getDNSrecords: "",
+        setDNSrecords: "",
+        domain: "",
+        address: "",
+      },
+    };
+  },
+  computed: {
+    ...mapState(["instanceName", "core", "appName"]),
+    i18nTableColumns() {
+      return this.tableColumns.map((column) => {
+        return this.$t("DNSrecords." + column);
+      });
+    },
+  },
+  beforeRouteEnter(to, from, next) {
+    next((vm) => {
+      vm.watchQueryData(vm);
+      vm.urlCheckInterval = vm.initUrlBindingForApp(vm, vm.q.page);
+    });
+  },
+  beforeRouteLeave(to, from, next) {
+    clearInterval(this.urlCheckInterval);
+    next();
+  },
+  created() {
+    this.getDNSrecords();
+  },
+  methods: {
+    async getDNSrecords() {
+      this.loading.getDNSrecords = true;
+      this.error.getDNSrecords = "";
+      const taskAction = "get-dns-records";
+      const eventId = this.getUuid();
+
+      // register to task error
+      this.core.$root.$once(
+        `${taskAction}-aborted-${eventId}`,
+        this.getDNSrecordsAborted
+      );
+
+      // register to task completion
+      this.core.$root.$once(
+        `${taskAction}-completed-${eventId}`,
+        this.getDNSrecordsCompleted
+      );
+
+      const res = await to(
+        this.createModuleTaskForApp(this.instanceName, {
+          action: taskAction,
+          extra: {
+            title: this.$t("action." + taskAction),
+            isNotificationHidden: true,
+            eventId,
+          },
+        })
+      );
+      const err = res[0];
+
+      if (err) {
+        console.error(`error creating task ${taskAction}`, err);
+        this.error.getDNSrecords = this.getErrorMessage(err);
+        this.loading.getDNSrecords = false;
+        return;
+      }
+    },
+    getDNSrecordsAborted(taskResult, taskContext) {
+      console.error(`${taskContext.action} aborted`, taskResult);
+      this.error.getDNSrecords = this.$t("error.generic_error");
+      this.loading.getDNSrecords = false;
+    },
+    getDNSrecordsCompleted(taskContext, taskResult) {
+      this.loading.getDNSrecords = false;
+      this.DNSRecords = taskResult.output.records;
+    },
+    validateDNSrecords() {
+      this.clearErrors(this);
+      let isValidationOk = true;
+
+      if (!this.domain) {
+        // test field cannot be empty
+        this.error.domain = this.$t("common.required");
+
+        if (isValidationOk) {
+          isValidationOk = false;
+        }
+      }
+      if (!this.address) {
+        // test field cannot be empty
+        this.error.address = this.$t("common.required");
+
+        if (isValidationOk) {
+          isValidationOk = false;
+        }
+      }
+
+      return isValidationOk;
+    },
+    setDNSvalidationFailed(validationErrors) {
+      this.loading.setDNSrecords = false;
+
+      for (const validationError of validationErrors) {
+        this.error.address = this.$t("error." + validationError.error);
+      }
+    },
+    async setDNSrecords(recordToDelete) {
```

Comment:
For clarity, let's add a comment that explains that this function in used for both record creation and deletion